Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2926,8 +2926,23 @@ extension BrowserViewController: IntroViewControllerDelegate {
         } else {
             let signInVC = FxAContentViewController(profile: profile)
             signInVC.delegate = self
-            signInVC.url = profile.accountConfiguration.signInURL
-            signInVC.fxaOptions = fxaOptions
+            
+            // Append any passed query strings to the signin url. Note that you can't
+            // override the service and context params.
+            if fxaOptions.query != nil {
+                var url = "";
+                for param in fxaOptions.query! {
+                    if (param.key == "service" || param.key == "context") {
+                        continue;
+                    }
+                    url = url + "&" + param.key + "=" + param.value
```

Comment:
This was misleading, it was actually building out the query strings to append onto the final signin URL. Updated variable name to reflect that it is query params and not url.